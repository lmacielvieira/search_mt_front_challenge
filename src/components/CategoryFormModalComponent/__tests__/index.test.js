import React from 'react'
import {shallow, mount} from 'enzyme'
import {ContactFormModalComponent} from '../index'

describe('<ContactDeleteModalComponent />', () => {
  it('renders <ContactFormModalComponent /> closed', () => {
    const wrapper = shallow(<ContactFormModalComponent />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders <ContactFormModalComponent /> opened', () => {
    const wrapper = shallow(<ContactFormModalComponent visible />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders <ContactFormModalComponent /> and expect to call handleCancelCb', () => {
    const onClose = jest.fn()
    const wrapper = shallow(
      <ContactFormModalComponent visible handleCancelCb={onClose} />
    )
    wrapper.find({'data-testid': 'CFMcancelButton'}).simulate('click')
    expect(onClose).toHaveBeenCalledTimes(1)
  })

  it('renders <ContactFormModalComponent /> and expect submit to be disabled', () => {
    const onClose = jest.fn()
    const wrapper = shallow(<ContactFormModalComponent visible />)
    const btn = wrapper.find({'data-testid': 'CFMsubmitButton'})
    btn.simulate('click')
    expect(btn.prop('disabled')).toBe(true)
    expect(onClose).not.toHaveBeenCalledTimes(1)
  })

  it('renders <ContactFormModalComponent /> and expect submit to be enabled after typing name', () => {
    const wrapper = mount(<ContactFormModalComponent visible />)
    wrapper
      .find({'data-testid': 'CFMnameButton'})
      .at(0)
      .simulate('change', {target: {value: 'Hello world'}})

    jest.useFakeTimers()
    setTimeout(() => {
      const btn = wrapper.find({'data-testid': 'CFMsubmitButton'}).at(0)
      expect(btn.prop('disabled')).toBe(false)
    }, 500)
    jest.runAllTimers()
  })
})
