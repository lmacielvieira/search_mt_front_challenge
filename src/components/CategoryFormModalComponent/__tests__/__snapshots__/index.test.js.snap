// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactDeleteModalComponent /> renders <ContactFormModalComponent /> closed 1`] = `ShallowWrapper {}`;

exports[`<ContactDeleteModalComponent /> renders <ContactFormModalComponent /> opened 1`] = `ShallowWrapper {}`;
